File: DriverTemplates/TemplateSources/src/ASCOM Driver Template (VB)/DeviceFocuser.vb
' All lines from line 1 to the device interface implementation region will be discarded by the project wizard when the template is used
' Required code must lie within the device implementation region
' The //ENDOFINSERTEDFILE tag must be the last but one line in this file

Imports ASCOM.DeviceInterface
Imports ASCOM
Imports ASCOM.Utilities

Class DeviceFocuser
    Implements IFocuserV2
    Private m_util As New Util()
    Private TL As New TraceLogger()

#Region "IFocuser Implementation"
    Public ReadOnly Property Absolute() As Boolean Implements IFocuserV2.Absolute
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public Sub Halt() Implements IFocuserV2.Halt
        Throw New ASCOM.MethodNotImplementedException()
    End Sub

    Public ReadOnly Property IsMoving() As Boolean Implements IFocuserV2.IsMoving
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public Property Link() As Boolean Implements IFocuserV2.Link
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
        Set(value As Boolean)
            Throw New ASCOM.PropertyNotImplementedException()
        End Set
    End Property

    Public ReadOnly Property MaxIncrement() As Integer Implements IFocuserV2.MaxIncrement
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public ReadOnly Property MaxStep() As Integer Implements IFocuserV2.MaxStep
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public Sub Move(Position As Integer) Implements IFocuserV2.Move
        Throw New ASCOM.MethodNotImplementedException()
    End Sub

    Public ReadOnly Property Position() As Integer Implements IFocuserV2.Position
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public ReadOnly Property StepSize() As Double Implements IFocuserV2.StepSize
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public Property TempComp() As Boolean Implements IFocuserV2.TempComp
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
        Set(value As Boolean)
            Throw New ASCOM.PropertyNotImplementedException()
        End Set
    End Property

    Public ReadOnly Property TempCompAvailable() As Boolean Implements IFocuserV2.TempCompAvailable
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

    Public ReadOnly Property Temperature() As Double Implements IFocuserV2.Temperature
        Get
            Throw New ASCOM.PropertyNotImplementedException()
        End Get
    End Property

#End Region

    '//ENDOFINSERTEDFILE
End Class

File: trunk/DriverTemplates/TemplateSources/src/ASCOM Driver Template (CS)/DeviceFocuser.cs
// All lines from line 1 to the device interface implementation region will be discarded by the project wizard when the template is used
// Required code must lie within the device implementation region
// The //ENDOFINSERTEDFILE tag must be the last but one line in this file

using ASCOM.DeviceInterface;
using System;
using ASCOM;
using ASCOM.Utilities;

class DeviceFocuser
{
    Util util = new Util(); TraceLogger tl = new TraceLogger();

    #region IFocuser Implementation
    public bool Absolute
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public void Halt()
    {
        throw new ASCOM.MethodNotImplementedException();
    }

    public bool IsMoving
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public bool Link
    {
        get
        {
            throw new ASCOM.PropertyNotImplementedException();
        }
        set
        {
            throw new ASCOM.PropertyNotImplementedException();
        }
    }

    public int MaxIncrement
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public int MaxStep
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public void Move(int Position)
    {
        throw new ASCOM.MethodNotImplementedException();
    }

    public int Position
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public double StepSize
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public bool TempComp
    {
        get
        {
            throw new ASCOM.PropertyNotImplementedException();
        }
        set
        {
            throw new ASCOM.PropertyNotImplementedException();
        }
    }

    public bool TempCompAvailable
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    public double Temperature
    {
        get { throw new ASCOM.PropertyNotImplementedException(); }
    }

    #endregion

    //ENDOFINSERTEDFILE
}